core/src/main/java/hudson/Functions.java

```diff
@@ -1064,7 +1066,7 @@ public static String getFooterURL() {
         List<Descriptor> answer = new ArrayList<>(r.size());
         for (Tag d : r) answer.add(d.d);
 
-        return DescriptorVisibilityFilter.apply(Jenkins.get(),answer);
+        return GlobalConfigHiddenByPermissionFilter.apply(Jenkins.get(),answer);
```

Comment:
Shouldn't this work with the `DescriptorVisibilityFilter`? It seems that what it does is taking all registered descriptors filters, so it will take the new filter as well.

Reply:
`DescriptorVisiblityFilter` is an abstract class with enough static methods for the original check to work.  All i have done here is overide the `Filter` method to check the permission, the default implementation in `DescriptorVisiblityFilter` always returns true.